.github/workflows/build-wheels-publish.yml

```diff
@@ -0,0 +1,303 @@
+name: Build wheels with cibuildwheel
+
+on:
+  workflow_dispatch:
+    inputs:
+      debugMode:
+        description: 'Debug Mode'
+        required: false
+        default: ''
+        type: choice
+        options:
+          - '0'
+          - '1'
+
+  push:
+    paths-ignore:
+      - ".gitignore"
+      - "CREDITS.TXT"
+      - "ChangeLog"
+      - "README.md"
+      - "docs/**"
+  pull_request:
+
+env:
+  # Enable DEBUG flag either according to the tag release or manual override
+  CAPSTONE_DEBUG: ${{ inputs.debugMode != '' && inputs.debugMode || startsWith(github.ref, 'refs/tags') && '0' || '1' }}
+
+jobs:
+  # job to be executed for every push - testing purpose
+  build_wheels_always:
+    name: Building on ${{ matrix.os }} - ${{ matrix.arch }} ${{ matrix.cibw_build }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # NOTE: Making this to parallelize and speed up workflow
+          # i686 - manylinux
+          - { os: ubuntu-latest, arch: i686, cibw_build: 'cp38-manylinux', cibw_skip: '' }
+          # i686 - musllinux
+          - { os: ubuntu-latest, arch: i686, cibw_build: 'cp38-musllinux', cibw_skip: '' }
+          # x86_64 - manylinux
+          - { os: ubuntu-latest, arch: x86_64, cibw_build: 'cp38-manylinux', cibw_skip: '' }
+          # x86_64 - musllinux
+          - { os: ubuntu-latest, arch: x86_64, cibw_build: 'cp38-musllinux', cibw_skip: '' }
+          # aarch64 - manylinux
+          - { os: ubuntu-latest, arch: aarch64, cibw_build: 'cp38-manylinux', cibw_skip: '' }
+          # aarch64 - musllinux
+          - { os: ubuntu-latest, arch: aarch64, cibw_build: 'cp38-musllinux', cibw_skip: '' }
+          # macos - x86_64
+          - { os: macos-13, arch: x86_64, cibw_build: 'cp38', cibw_skip: '' }
+          # macos - arm64
+          - { os: macos-latest, arch: arm64, cibw_build: 'cp38', cibw_skip: '' }
+          # windows - x86_64
+          - { os: windows-latest, arch: AMD64, cibw_build: 'cp38', cibw_skip: '' }
+          # windows - amd64
+          - { os: windows-latest, arch: x86, cibw_build: 'cp38', cibw_skip: '' }
+          # windows - arm64
+          - { os: windows-latest, arch: ARM64, cibw_build: 'cp39', cibw_skip: '' }
```

Comment:
What I wanted to achieve here is leverage as much as possible the matrix parallelism of Github runners, rather having cibuildwheel single plan dealing with multiple versions/platforms. Beside speeding-up aspect, it should also be able to spot issues without stopping at first failed build: for instance if we want to build for manylinux and musllinux, in case manylinux fails we will not see if it will also fail for musllinux since it will exit. I do understand the concern about quota so once we decide how to go ahead I'm available to adapt accordingly.

Regarding the python version basically what I did is to build and test for each push/PR only the lowest python version and all the rest in case of tag release (for a matter of time). You see python 3.9 only for the Windows ARM64 build since cibuildwheel has no py3.8 target for it, that is why. The approach of using lowest/higher I think it makes sense and I can do the change. Even though I would keep the building of the other versions in case of tag release to be sure the wheels uploaded to Pypi are not broken.